The-SMIRNOFF-force-field-format.md

```diff
@@ -138,54 +332,61 @@ Generalized-Born surface area (GBSA) implicit solvent parameters can also be spe
+* `OBC2`: [Onufriev-Bashford-Case](http://docs.openmm.org/latest/userguide/zbibliography.html#onufriev2004) using the GB(OBC)II parameters (corresponding to `igb=5` in AMBER): requires `[radius, scale]`
+
+If the `gb_model` attribute is omitted, it defaults to `OBC1`.
 
 Each GB model can possess several attributes, which may be unitless (`1.0`, `78.5`) or unit-bearing quantities (`1.4*angstrom`, `5.4*calories/mole/angstrom**2`).
 The attributes `solvent_dielectric` and `solute_dielectric` specify solvent and solute dielectric constants used by the GB model.
 In this example, `radius` and `scale` are per-particle parameters of the `OBC1` GB model supported by OpenMM.
 Units are for these per-particle parameters (such as `radius_units`) optionally specified in the `<GBSAForce/>` tag.
 
-#### SA model
+#### Surface area (SA) penalty model
 
 The `sa_model` attribute specifies the solvent-accessible surface area model ("SA" part of GBSA) if one should be included; if omitted, no SA term is included.
 
-Currently, only the [analytical continuum electrostatics (ACE) model](http://docs.openmm.org/7.1.0/userguide/theory.html#surface-area-term), designated `ACE`, can be specified, but there are plans to add more models in the future, such as the Gaussian solvation energy component of [EEF1](https://www.ncbi.nlm.nih.gov/pubmed/10223287).
+Currently, only the [analytical continuum electrostatics (ACE) model](http://docs.openmm.org/latest/userguide/theory.html#surface-area-term), designated `ACE`, can be specified, but there are plans to add more models in the future, such as the Gaussian solvation energy component of [EEF1](https://www.ncbi.nlm.nih.gov/pubmed/10223287).
+If `sa_model` is not specified, it defaults to `ACE`.
+
 The `ACE` model permits two additional parameters to be specified:
-* The `surface_area_penalty` attribute specifies the surface area penalty for the `ACE` model (defaults to `5.4*calories/mole/angstroms**2`).
-* The `solvent_radius` attribute specifies the solvent radius, which defaults to `1.4*angstroms`.
+* The `surface_area_penalty` attribute specifies the surface area penalty for the `ACE` model. (Default: `5.4*calories/mole/angstroms**2`)
+* The `solvent_radius` attribute specifies the solvent radius. (Default: `1.4*angstroms`)
 
-### SPECIAL SECTIONS
+### `<BondChargeCorrections>`
 
-**Bond charge corrections**
-Bond charge corrections (along the lines of Christopher Bayly's bond charge corrections in [AM1-BCC](https://dx.doi.org/10.1002/jcc.10128)) can be applied via a `BondChargeCorrections` tag with children specifying specific `BondChargeCorrection` terms.
+Bond charge corrections similar to those used in the [AM1-BCC](https://dx.doi.org/10.1002/jcc.10128) charge model from Christopher Bayly can be applied via a `<BondChargeCorrections>...</BondChargeCorrections>` section with child tags specifying specific `<BondChargeCorrection/>` terms.
```

Comment:
Should this part not be tied more closely with library charges? If you already have library charges that you calculated with a method that incorporates BCCs, for example, does this override or do something different? I think I understand each section (library charges and BCCs) on their own, but how are they related? And if the user supplies their own charges using whatever method, can they make sure no BCCs from the force field are applied, even if their molecule matches the pattern, e.g. `[#6X4:1]-[#6X3a:2]` in the `BondChargeCorrections` section of the default `ffxml`?

Reply:
Good point. The idea was that library charges would have highest priority in matching entire molecules or residues, and other schemes would be used for unmatched molecules. I should explicitly state this.

If multiple residue templates match, should the last one win? If it's a subset of another residue match, what should happen?

If we do have multiple schemes in the future, such as AM1-BCC and RESP schemes, should both be allowed at once? Or should there be priority based on order in which these appear?

Reply:
I like the idea that library charges have highest priority. If multiple templates match, it should be last-one-wins like the rest of the force field, because doing anything else would be confusing.

>If we do have multiple schemes in the future, such as AM1-BCC and RESP schemes, should both be allowed at once? Or should there be priority based on order in which these appear?

I think we would want to raise an exception if both are provided because I don't see how it would make sense and would be complicated to support in a reasonable way.

David S. wrote this, which also seems important:
> And if the user supplies their own charges using whatever method, can they make sure no BCCs from the force field are applied

Maybe we need a clear hierarchy with respect to charges, AND an option relating to input charges, where a user would ask `createSystem` to use user-supplied charges (`user_supplied=True` maybe). Perhaps:
1) Library charges always win, and are assigned regardless of whether the user asks for user-supplied charges
2) If user supplied charges are requested via `user_supplied=True`, no BCCs or other charge calculations are done; only library charges are applied
3) If `user_supplied=False`, BCCs and other charge methods are applied as dictated by FF.

It's important to be clear about when user-supplied charges would be applied and when not. Does that sound like it would work well, @slochower ?

Reply:
Yes! The 3 step process looks good!